components/brave_news/browser/brave_news_controller.cc

```diff
@@ -57,6 +58,9 @@ namespace {
 // The favicon size we desire. The favicons are rendered at 24x24 pixels but
 // they look quite a bit nicer if we get a 48x48 pixel icon and downscale it.
 constexpr uint32_t kDesiredFaviconSizePixels = 48;
+// Since we have two boolean prefs for the News enabled status, a delay
+// will be used so that we only report the histogram once for both pref updates.
```

Comment:
What's the problem with reporting the histogram twice - won't it just report only the most recent value for this metric?

Reply:
not quite, it seems that when a user presses "no thanks" on Android, both prefs will be enabled in the first call, and then the "show today" pref will be disabled in the second call. if we don't have the timer, the first call will set the "was enabled" pref to true which will result in the "is enabled" metric being reported as 0 in the second call, which we don't want. the timer is here to make sure all edge cases are covered and increases robustness

Reply:
This seems like a bug from Android. We should locate that part @deeppandya @tapanmodh ?

Reply:
@petemill sure. I'll check it